packages/block-editor/src/hooks/background.js

```diff
@@ -0,0 +1,271 @@
+			},
+		} ),
+	} );
+}
+
+function InspectorImagePreview( { url: imgUrl } ) {
+	const imgLabel = getFilename( imgUrl );
+	return (
+		<ItemGroup as="span">
+			<HStack justify="flex-start" as="span">
+				<img src={ imgUrl } alt="" />
+				<FlexItem as="span">
+					<Truncate
+						numberOfLines={ 1 }
+						className="block-editor-hooks__background__inspector-media-replace-title"
+					>
+						{ imgLabel }
+					</Truncate>
+				</FlexItem>
+			</HStack>
+		</ItemGroup>
+	);
+}
+
+function BackgroundImagePanelItem( props ) {
+	const { attributes, clientId, setAttributes } = props;
+
+	const { id, url } = attributes.style?.background?.backgroundImage || {};
+
+	const { mediaUpload } = useSelect( ( select ) => {
+		return {
+			mediaUpload: select( blockEditorStore ).getSettings().mediaUpload,
+		};
+	} );
+
+	const { createErrorNotice } = useDispatch( noticesStore );
+	const onUploadError = ( message ) => {
+		createErrorNotice( message, { type: 'snackbar' } );
+	};
+
+	const onSelectMedia = ( media ) => {
+		if ( ! media || ! media.url ) {
+			const newStyle = {
+				...attributes.style,
+				background: {
+					...attributes.style?.background,
+					backgroundImage: undefined,
+				},
+			};
+
+			const newAttributes = {
+				style: cleanEmptyObject( newStyle ),
+			};
+
+			setAttributes( newAttributes );
+			return;
+		}
+
+		if ( isBlobURL( media.url ) ) {
+			// TODO: Might need to revoke the blob URL.
+			return;
+		}
+
+		// For media selections originated from a file upload.
+		if (
+			( media.media_type &&
+				media.media_type !== IMAGE_BACKGROUND_TYPE ) ||
+			( ! media.media_type &&
+				media.type &&
+				media.type !== IMAGE_BACKGROUND_TYPE )
+		) {
+			onUploadError(
+				__( 'Only images can be used as a background image.' )
+			);
+			return;
+		}
+
+		const newStyle = {
+			...attributes.style,
+			background: {
+				...attributes.style?.background,
+				backgroundImage: {
+					url: media.url,
+					id: media.id,
+					source: 'file',
+				},
+			},
+		};
+
+		const newAttributes = {
+			style: cleanEmptyObject( newStyle ),
+		};
+
+		setAttributes( newAttributes );
+	};
+
+	const onFilesDrop = ( filesList ) => {
+		mediaUpload( {
+			allowedTypes: [ 'image' ],
+			filesList,
+			onFileChange( [ image ] ) {
+				if ( isBlobURL( image?.url ) ) {
+					return;
+				}
+				onSelectMedia( image );
+			},
+			onError: onUploadError,
+		} );
+	};
+
+	const resetAllFilter = useCallback( ( previousValue ) => {
+		return {
+			...previousValue,
+			style: {
+				...previousValue.style,
+				background: undefined,
+			},
+		};
+	}, [] );
+
+	return (
+		<ToolsPanelItem
+			className="single-column"
+			hasValue={ () => hasBackgroundImageValue( props ) }
+			label={ __( 'Background image' ) }
+			onDeselect={ () => resetBackgroundImage( props ) }
+			isShownByDefault={ true }
+			resetAllFilter={ resetAllFilter }
+			panelId={ clientId }
+		>
+			<div className="block-editor-hooks__background__inspector-media-replace-container">
```

Comment:
I expected this to be a reused component with the site logo. I guess it's easy to build, but we may end up with different looking controls in time.

Reply:
Yes, I thought we could look at consolidating the components once this PR lands and we feel confident about the UI being consistent. I figured it'd be a bit easier to keep them separate while we introduce the new support, and then could look at building a shared component in a follow-up.